extra_data/reader.py

```diff
@@ -1013,6 +1015,12 @@ def select(self, seln_or_source_glob, key_glob='*', require_all=False,
                     else:  # require_any
                         train_ids = np.union1d(train_ids, source_tids)
 
+                if len(train_ids) < (n_trains_prev / 2):
```

Comment:
How about a configurable threshold via keyword parameter? Then this feature can be adapted to any situation and actively used to test against expectations. `0.5` could be the default value.

Reply:
Thinking about it more after commmitting, I might make the default 0% left (= 100% dropped). The most common case for this is a source that recorded no data at all, so this would still catch that, but avoid any annoying warnings when they're not wanted.